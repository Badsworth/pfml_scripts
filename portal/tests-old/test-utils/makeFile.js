/**
 * Make a File instance
 * @see https://developer.mozilla.org/en-US/docs/Web/API/File
 * @param {object} attrs File attributes
 * @param {string} attrs.name The file's name
 * @param {string} attrs.type The file's MIME type
 * @returns {File}
 */
export const makeFile = (attrs = {}) => {
  const { name, type } = Object.assign(
    {
      name: "file.pdf",
      type: "application/pdf",
    },
    attrs
  );

  return new File([], name, { type });
};

/**
 * Create an HTML <input> for a test's event.target
 * @param {object} attrs - HTML attributes for the input
 * @returns {HTMLInputElement}
 */
export function createInputElement(attrs) {
  const input = document.createElement("input");

  Object.entries(attrs).forEach(([attributeName, value]) => {
    input.setAttribute(attributeName, value);
  });

  if (attrs.checked === false) input.checked = false;

  return input;
}
