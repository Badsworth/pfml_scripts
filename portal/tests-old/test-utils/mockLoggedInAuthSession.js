import { Auth } from "@aws-amplify/auth";

/**
 * Mock the Cognito authentication session so a test user appears logged in
 */
const mockLoggedInAuthSession = () => {
  const accessTokenJwt =
    "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJuYW1lIjoiQnVkIn0.YDRecdsqG_plEwM0H8rK7t2z0R3XRNESJB5ZXk-FRN8";

  jest.spyOn(Auth, "currentSession").mockImplementation(() =>
    Promise.resolve({
      accessToken: { jwtToken: accessTokenJwt },
    })
  );
};

export default mockLoggedInAuthSession;
