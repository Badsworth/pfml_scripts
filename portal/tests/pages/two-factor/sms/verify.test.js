import { act, screen } from "@testing-library/react";
import { mockAuth, renderPage } from "../../../test-utils";
import VerifySMS from "../../../../src/pages/two-factor/sms/verify";
import userEvent from "@testing-library/user-event";

jest.mock("../../../../src/services/tracker");

beforeEach(() => {
  mockAuth(true);
  process.env.featureFlags = { claimantShowMFA: true };
});

describe("Two-factor SMS Verify", () => {
  it("renders landing page content", () => {
    const { container } = renderPage(VerifySMS);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders PageNotFound if the claimantShowMFA feature flag is not set", () => {
    process.env.featureFlags = { claimantShowMFA: false };
    renderPage(VerifySMS);

    const pageNotFoundHeading = screen.getByRole("heading", {
      name: /Page not found/,
    });
    expect(pageNotFoundHeading).toBeInTheDocument();
  });

  it("sends verification code and updates MFA preference when user saves and continues", async () => {
    const mockVerify = jest.fn();
    renderPage(
      VerifySMS,
      {
        addCustomSetup: (appLogic) => {
          appLogic.auth.verifyMFACodeAndLogin = mockVerify;
        },
      },
      { query: { next: "next" } }
    );

    const codeField = screen.getByLabelText("6-digit code");
    userEvent.type(codeField, "123456");
    const submitButton = screen.getByRole("button", {
      name: "Submit",
    });
    await act(async () => await userEvent.click(submitButton));

    expect(mockVerify).toHaveBeenCalledWith("123456", "next");
  });
});
