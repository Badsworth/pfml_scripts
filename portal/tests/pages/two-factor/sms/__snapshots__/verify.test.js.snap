// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Two-factor SMS Verify renders landing page content 1`] = `
<div>
  <form
    class="usa-form"
    method="post"
  >
    <h1
      class="js-title margin-top-0 margin-bottom-2 font-heading-lg line-height-sans-2"
      tabindex="-1"
    >
      Enter your security code
    </h1>
    <p
      class="usa-intro"
    >
      We sent a 6-digit code to your phone number ending in 6789.
    </p>
    <div
      class="usa-form-group"
    >
      <label
        class="usa-label text-bold font-heading-xs measure-5"
        for="InputText1"
        id="InputText1_label"
      >
        6-digit code
      </label>
      <input
        aria-labelledby="InputText1_label InputText1_hint InputText1_error"
        autocomplete="one-time-code"
        class="usa-input maxw-mobile-lg"
        id="InputText1"
        inputmode="numeric"
        name="code"
        type="text"
        value=""
      />
    </div>
    <button
      class="usa-button position-relative display-block margin-top-1 usa-button--unstyled"
      type="button"
    >
      Resend the code
    </button>
    <button
      class="usa-button position-relative display-block margin-top-3"
      type="submit"
    >
      Submit
    </button>
  </form>
  <p
    class="display-block margin-top-3"
  >
    Need help? Call the Contact Center at 
    <a
      href="tel:(833) 344-7365"
    >
      (833) 344‑7365
    </a>
    .
  </p>
</div>
`;
